File: LeetCode/C++/26. Remove Duplicates from Sorted Array.cpp
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        int uniqueTotal = 1;
        int currentUnique = nums[0];
        int uniqueIndex = 0;

        for (int i = 1; i < nums.size(); i++)
        {
            if (nums[i] != currentUnique)
            {
                currentUnique = nums[i];
                nums[++uniqueIndex] = nums[i];
                uniqueTotal++;
            }
        }

        return uniqueTotal;
    }
};

File: LeetCode/Top Interview 150 - Java/26. Remove Duplicates from Sorted Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 1;
        int currentNum = nums[0];

        for(int i = 1; i < nums.length; i++) {
            if(nums[i] != currentNum) {
                nums[k] = nums[i];
                currentNum = nums[i];
                k++;
            }
        }

        return k;
    }
}